samza-core/src/test/java/org/apache/samza/operators/TestJoinOperator.java

```diff
@@ -71,8 +71,27 @@ public void join() throws Exception {
   }
 
   @Test
+  public void testJoinFnInitAndClose() throws Exception {
+    TestJoinFunction joinFn = new TestJoinFunction();
+    StreamOperatorTask sot = createStreamOperatorTask(new SystemClock(), new TestJoinStreamApplication(joinFn));
+    List<Integer> output = new ArrayList<>();
+    MessageCollector messageCollector = envelope -> output.add((Integer) envelope.getMessage());
```

Comment:
The `messageCollector` is not needed. Are you referring to that? 
We should probably still retain calls to `sot.process()` to ensure that `process()` does n't accidentally end up closing operators?